traits/has_traits.py

```diff
@@ -2034,7 +2039,11 @@ def configure_traits(
                 with open(filename, "rb") as fd:
                     self.copy_traits(pickle.Unpickler(fd).load())
 
-        if edit:
+        if edit is True or None:
```

Comment:
This line probably isn't doing what you think it's doing.  It's implicitly bracketed as `if edit is (True or None)`, and `True or None` is `True`, so it's equivalent to `if edit is True`.

Please  reconsider my earlier suggestion to just add a block:

```python
if edit is None:
    edit = True
else:
     message = (... as before ...)
     warnings.warn(message, DeprecationWarning)
```

and leave the rest of the logic unaltered. This covers all the necessary cases, without any repetition of key logic.

Reply:
ok @mdickinson , I'll do so.